### PR TITLE
Migration to mkdocs

### PR DESCRIPTION
**Pull request by abhi18av**
Here's the initial feature-complete implementation for the `mkdocs` migration.

To build this version

```
make pull

make serve

```

Here's how it looks

### CODE REVIEW

**Comment by pditommaso** [comment]
It looks very good. Couple of things: 

1. Is there a way to get rid of the black bottom navigation bar? 
2. How to configure/manage docs versioning? 
3. Found some warnings 

> INFO    -  Browser Connected: http://localhost:8000/
[W 210413 10:14:39 web:2239] 404 GET /assets/images/uploads/2020/10/actions_tower_hook_params.png (172.17.0.1) 138.44ms
WARNING -  404 GET /assets/images/uploads/2020/10/actions_tower_hook_params.png (172.17.0.1) 138.44ms
[I 210413 10:15:15 handlers:135] Browser Connected: http://localhost:8000/
INFO    -  Browser Connected: http://localhost:8000/
[W 210413 10:16:03 web:2239] 404 GET /assets/images/uploads/2020/10/actions_tower_hook_params.png (172.17.0.1) 8.10ms
WARNING -  404 GET /assets/images/uploads/2020/10/actions_tower_hook_params.png (172.17.0.1) 8.10ms
[W 210413 10:17:09 web:2239] 404 GET /assets/images/uploads/2020/10/actions_tower_hook_params.png (172.17.0.1) 3.94ms
WARNING -  404 GET /assets/images/uploads/2020/10/actions_tower_hook_params.png (172.17.0.1) 3.94ms

**Comment by abhi18av** [comment]
Thanks for the review :) 

>    Is there a way to get rid of the black bottom navigation bar?

Yes, I'm working on it right now. [Here's the docs](https://squidfunk.github.io/mkdocs-material/customization/) I'm referring.

>    How to configure/manage docs versioning?

Yes, it is possible. I have created a sample over here  https://abhi18av.github.io/nf-tower-docs-mkdocs/21.03.5/ 
The repo is here https://github.com/abhi18av/nf-tower-docs-mkdocs, notice the structure of the repo.

**NOTE**:  The downside is that this works only with the `gh-pages` branch - which could be used to serve to `help.tower.nf` as well via repo settings.

>    Found some warnings

Thanks for pointing out, I have fixed those.

**Comment by abhi18av** [comment]
Okay, regarding the `navigation bar`, here's how the website looks after removal

**Comment by pditommaso** [comment]
> The downside is that this works only with the gh-pages branch - which could be used to serve to help.tower.nf as well via repo settings.

Oh really!? why?

**Comment by abhi18av** [comment]
The reason is that `mkdocs-material` [implements versioning feature](https://squidfunk.github.io/mkdocs-material/setup/setting-up-versioning/) with an external tool called [`mike` ](https://github.com/jimporter/mike) which explitely works with `gh-pages` branch.

Generally speaking, `gh-pages` might be enough for the docs. What do you think?

**Comment by abhi18av** [comment]
Update, if you'd like to see how the `versioning` feature would work out here's the sample https://abhi18av.github.io/nf-tower-docs-mkdocs/latest/

NOTE: The older versions mentioned here are dev time versions I pushed to test the feature.

If this looks okay, let me know :)

**Comment by pditommaso** [comment]
That's nice but it's really bad we should move to gh pages. Looks very such a hard dependency. In the doc it mentions `provider:` maybe it can be implemented a generic one? can you investigate about that?

**Comment by abhi18av** [comment]
Yup, I can give it a closer look.

In the meantime, I have identified an issue with `versioned` docs, the `mike` tool doesn't update the links for the `image` accordingly to how it structures various docs together.

When using a simpe unversioned doc, the images are rendered just fine.

**Comment by abhi18av** [comment]
Okay, so I have experimented more and read the codebase through which I came to know about the undocumented feature of allowing any `branch` as the branch for publishing various versions of the docs.

https://github.com/jimporter/mike/blob/d60c3812b03cdbe2369061f5b1df7692302e6c99/mike/mkdocs_utils.py#L18

The other point being, to use the versioning feature we need to use the `mike serve` instead of the `mkdocs serve` feature.

**Comment by abhi18av** [comment]
Another issue is that using `mike serve` the `search` functionality is not available, whereas the vanilla `mkdocs serve` it is clearly visible 

image

**Comment by pditommaso** [comment]
let's keep the versioning as an improvement and focus on current version migration

**Comment by abhi18av** [comment]
Cool, I've accomodated the code review and managed to fix the problem with search box.

The latest version is available here https://seqeralabs.github.io/nf-tower-docs/21.04.0/ 

This PR is all set from my side, please let me know if I need to do anything improvements here :)

**Review by pditommaso — CHANGES_REQUESTED** [review]
Looks good! just a minor change to update the README with the new build commands. Also the `Seqera Labs` in the footer cannot be made a link to https://seqera.io ?

**Comment by abhi18av** [comment]
Cool, I've updated the footer link for the main Seqera website 👍

**Comment by pditommaso** [comment]
Please update also the readme.

**Comment by pditommaso** [comment]
Also, we should add the GA tracking 

```
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-364526-12"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'UA-364526-12');
</script>
```

Not sure if the HTML meta was set 

```
<meta name="description" content="Nextflow Tower is a full-stack application for the development, deployment and monitoring of Nextflow data analysis pipelines. /">
```

One more thing, can you check how to use the same font as the current on help.tower.nf

**Comment by abhi18av** [comment]
- [x] Please update also the readme.
- [x] Also, we should add the GA tracking , reference https://github.com/squidfunk/mkdocs-material/issues/2055#issuecomment-730974196
- [x] HTML meta

**Comment by pditommaso** [comment]
Re: font.  The browser says `font-family: "Avenir", "Helvetica Neue", Helvetica, sans-serif;` not understanding from where is loaded.

**Comment by abhi18av** [comment]
Hey @olehpratsko , could you please take a look into this one. We'd like to have similar fonts in the current hugo based docs on help.tower.nf/ and in the new version in this PR.

I'd be happy to connect to explain the context or to navigate the codebase for the theme.

**Comment by olehpratsko** [comment]
Hi @abhi18av, sorry for the delay, I can help you with the issue. Let me know when you're available to connect to clarify the details.

**Comment by abhi18av** [comment]
@olehpratsko 

Basically, we'd like to change the fonts in the new iteration here https://abhi18av.github.io/nf-tower-docs-orgs-and-teams/21.04.temp7/orgs-and-teams/overview/ to look similar to the ones used here https://help.tower.nf/

Here's a relevant comment thread https://github.com/seqeralabs/nf-tower-docs/pull/66#issuecomment-822480273 


I'd be happy to connect if you need more context.

**Comment by pditommaso** [comment]
Was checking this, what's the need for a custom container? cannot the standard mkdocs one used ?

**Comment by abhi18av** [comment]
We could, but for the versioned docs we needed to simply add `mike` dependency to the mix.

Now, `make versioned` will serve the versioned docs.

**Comment by pditommaso** [comment]
`mike` was not only working for GitHub pages?

**Comment by abhi18av** [comment]
I think it's possible to customize. Just that the features are undocumented 

> Okay, so I have experimented more and read the codebase through which I came to know about the undocumented feature of allowing any branch as the branch for publishing various versions of the docs.

> https://github.com/jimporter/mike/blob/d60c3812b03cdbe2369061f5b1df7692302e6c99/mike/mkdocs_utils.py#L18

x-ref https://github.com/seqeralabs/nf-tower-docs/pull/66#issuecomment-819134957

**Comment by pditommaso** [comment]
Ok, let's close this without caring about version, therefore using the standard container. 

Then, let's make a separate PR to handle the versioning with the custom container and mike, adding a `docker` folder with the Dockerfile to build the custom container

**Comment by abhi18av** [comment]
Cool, I've separate out the code related to versioning and finalized the docs for this version 👍

**Comment by pditommaso** [comment]
Can you confirm that `make build` create all HTML static assets in the `public` directory. Also let's move *Installation* (if not remove it)

**Comment by abhi18av** [comment]
By default, it publishes in the `site` directory but this can be customized using the `site_dir` variable https://www.mkdocs.org/user-guide/configuration/#site_dir 


Please let me know which one you prefer?

**Comment by pditommaso** [comment]
The publish task expected all the content web site content to be created in the `public` directory. 

https://github.com/seqeralabs/nf-tower-docs/blob/abhinav/mkdocs-migration/Makefile#L14

**Comment by abhi18av** [comment]
Done :)

File: mkdocs.yml [hunk 1]
```diff
@@ -104,10 +102,16 @@ nav:
       - 'Public Git repositories': 'git/public_repositories.md'
     - 'Automation':
       - 'Pipeline Actions': 'pipeline-actions/pipeline-actions.md'
+<<<<<<< HEAD
+=======
+    - 'Installation':
+      - 'System deployment': 'installation/system-deployment.md'
+>>>>>>> Evan review of teams
```
Comment (abhi18av) [inline mkdocs.yml:109]: @evanfloden , could you please clarify this one. I'm not sure which chunk you'd like to keep 🤔

**Comment by abhi18av** [comment]
@evanfloden I've completed my review of the content changes. Nice screenshot of the launchpad 🤩 

I've updated the broken links pointing to the original `/lauch/overiew` to `/launch/launch` file. 

As far as I can tell, we are good to go ✅  🚂

**Comment by pditommaso** [comment]
Now, let's break docs as well!

**Comment by abhi18av** [comment]
> Now, let's break docs as well!

😄 hehe

Hopefully not, I'm on the watchout. I'll ask a couple friends to review and suggest anything that we can improve :)

**Pull request merged**